appsTransport.go

```diff
@@ -75,6 +79,7 @@ func (t *AppsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	req.Header.Set("Authorization", "Bearer "+ss)
 	req.Header.Set("Accept", acceptHeader)
 
-	resp, err := t.tr.RoundTrip(req)
+	c := t.Client.(*http.Client)
+	resp, err := ctxhttp.Do(t.ctx, c, req)
```

Comment:
Can you explain why the `RoundTrip` call is being replaced by a `Do`?

This seems suspicious to me, because `Do` implements higher-level behavior than `RoundTrip`. Notably, https://godoc.org/net/http#RoundTripper says:

> RoundTrip executes a single HTTP transaction, returning
> a Response for the provided Request.
>
> RoundTrip should not attempt to interpret the response. In
> particular, RoundTrip must return err == nil if it obtained
> a response, regardless of the response's HTTP status code. ...

Whereas `Do`'s behavior is to follow redirects, etc.